duktape/src/main/jni/java/JavaType.cpp

```diff
@@ -178,11 +237,41 @@ struct Integer : public Primitive {
     return value;
   }
 
+  jarray popArray(duk_context* ctx, JNIEnv* env, uint32_t count, bool inScript) const {
+    jintArray array = env->NewIntArray(count);
+    for (auto i = count; i > 0u; --i) {
```

Comment:
why iterate over [count..1] and insert at [count-1..0] ? Wouldn’t it be simpler to iterate [count-1..0] ?